### DISCUSSION TITLE
Why was the "Extract Folder's Packages" feature removed in 4.4.3.6?

### DISCUSSION THREAD

**Discussion by nomakewan** [4sval/FModel#480] [Q&A] [answered]
I noticed after updating to 4.4.3.6 that the "Extract Folder's Packages" feature was removed from the right-click context menu. This feature was incredibly useful for, for example, extracting all the WEM audio files inside a directory so they could be operated on externally. After all, as of right now FModel does not seem to have any batch method of exporting all Wwise Audio as WAV.

For what reason was this feature removed? It seems to have been done in commit 21cfa07 which was just titled "bye bye" and offered zero explanation or justification for the removal.

Could this be either clarified or re-included in the next version, please? Thank you.

**Comment by 4sval** [comment, marked as answer]
This feature was misused by way too many people who thought it would extract whatever they needed, when all it was doing was parsing every single package in the directory and subdirectories, instead of extracting something.

It used to be useful back when this was a thing
![image](https://github.com/4sval/FModel/assets/26126862/cd8e10d4-214f-4a07-9343-2d789027ad55)

if you need anything in a directory you can still <kbd>CTRL</kbd><kbd>A</kbd> the packages tab and <kbd>SHIFT</kbd><kbd>DOUBLE LMB</kbd> to parse all packages or <kbd>RMB</kbd> for more options.
if you needed all the WEM files in a directory to work with them externally, `Extract Folder's Packages` was not the way to go anyway.

**Reply by nomakewan** [reply]
I appreciate the response.

Might I ask what the proper way to go to dump all the WEM files with FModel would have been, then?

**Reply by 4sval** [reply]
Export Raw Data

**Reply by nomakewan** [reply]
Aha. Then just delete all the useless uasset files and rename all the ubulk files to wem for processing. Gotcha. Slightly more roundabout but at least that works in the latest version. Thank you!

Honestly it would be rather sweet if we could have a batch WAV export function in FModel anyway. it's not that I necessarily would want WEMs, it's just that the fastest way to get all the files as WAV as of now is to dump the wems then convert them to WAV with vgmstream. Skipping that step would be wonderful if it's possible.

**Reply by 4sval** [reply]
WEM audio files have the explicit .wem extension in FModel.

If you encounter AkMediaAssetData objects type inside uassets you should <kbd>DOUBLE LMB</kbd>, let the audio player opens, and save the playlist. The first few WEMs will be converted to WAVs because it will auto play them, and all the others will have to be manually converted if not played inside FModel.
But never should you rename a ubulk to wem, it won't work if the data is prefetched.

**Reply by nomakewan** [reply]
That seems rather roundabout. Rather than just right-clicking the media directory and clicking "Extract Folder's Packages," instead I would open the media directory, select all the contents of the directory (`CTRL` `A` dumps you back to the Folders tab btw, it doesn't select all contents of the `Packages` tab), make sure auto-open audio files is enabled, open the selected files, wait for the parser to finish parsing all the uasset files, then click "Save playlist" to save most of the WEMs except for the one that got auto-played at the start of the process.

For giggles I timed the difference between these two operations when it came to exporting roughly 15k audio files from the exact same game.

4.4.3.5 "Extract Folder's Packages": 52 seconds (and ~1.5GB additional RAM usage during extraction)
4.4.3.6 "Save Playlist": Stopped experiment at 52 minutes having only extracted 2869 files (and ~10GB additional RAM usage during extraction)

The latter is insanely slow and memory-intensive despite the end result being identical. I find it hard to believe that this would be the "correct" method for batch exporting audio.

**Reply by 4sval** [reply]
This is not the method for batch exporting audio files, there's no method for batch exporting audio files.
FModel was made to explore archives, not unpack or batch export thousands of files for the purpose of whoever knows what.
If people wants to actually do that, we will always recommend them to use [CUE4Parse](https://github.com/FabianFG/CUE4Parse)

If you don't want to listen to a single audio file you can just disable `Auto Open Sounds` and it will auto save them without opening the audio player. Which is probably what you did with `Extract Folder's Packages` because this and selecting all the packages in a directory is literally the same, except that it doesn't deal with subdirs.

**Reply by nomakewan** [reply]
Okay, now I'm seeing where the disconnect is.

I was using a function of FModel that did what I needed but was unintended behavior. Now that that function is removed, there is no way to perform the task I needed using FModel.

Understood. Thank you once again for your prompt responses, I do sincerely appreciate it.
